tests/unit/jobs/test_views.py

```diff
@@ -371,3 +371,113 @@ def test_get_challenge_submissions(self):
         response = self.client.get(self.url, {})
         self.assertEqual(response.data['results'], expected)
         self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+
+class GetRemainingSubmissionTest(BaseAPITestClass):
+
+    def setUp(self):
+        super(GetRemainingSubmissionTest, self).setUp()
+        self.url = reverse_lazy('jobs:get_remaining_submissions',
+                                kwargs={'challenge_phase_id': self.challenge_phase.pk,
+                                        'challenge_id': self.challenge.pk})
+
+        self.submission1 = Submission.objects.create(
+            participant_team=self.participant_team,
+            challenge_phase=self.challenge_phase,
+            created_by=self.challenge_host_team.created_by,
+            status='submitted',
+            input_file=self.challenge_phase.test_annotation,
+            method_name="Test Method",
+            method_description="Test Description",
+            project_url="http://testserver/",
+            publication_url="http://testserver/",
+            is_public=True,
+        )
+
+        self.submission2 = Submission.objects.create(
+            participant_team=self.participant_team,
+            challenge_phase=self.challenge_phase,
+            created_by=self.challenge_host_team.created_by,
+            status='failed',
+            input_file=self.challenge_phase.test_annotation,
+            method_name="Test Method",
+            method_description="Test Description",
+            project_url="http://testserver/",
+            publication_url="http://testserver/",
+            is_public=True,
+        )
+
+    def test_get_remaining_submission_when_challenge_does_not_exist(self):
+        self.url = reverse_lazy('jobs:get_remaining_submissions',
+                                kwargs={'challenge_phase_pk': self.challenge_phase.pk,
+                                        'challenge_pk': self.challenge.pk})
+
+        expected = {
+            'detail': 'Challenge {} does not exist'.format(self.challenge.pk)
+        }
+
+        self.challenge.delete()
```

Comment:
I dont think there is a need to delete this challenge. Simply ask for `self.challenge.pk + 1` in the url, I think it will be sufficient.